evalml/pipelines/classification_pipeline.py

```diff
@@ -101,9 +101,11 @@ def predict(self, X, objective=None, X_train=None, y_train=None):
         Returns:
             pd.Series: Estimated labels.
         """
-        predictions = self._predict(X, objective=objective)
-        predictions = self.inverse_transform(predictions.astype(int))
-        predictions = pd.Series(predictions, name=self.input_target_name)
+        _predictions = self._predict(X, objective=objective)
+        predictions = self.inverse_transform(_predictions.astype(int))
+        predictions = pd.Series(
+            predictions, name=self.input_target_name, index=_predictions.index
```

Comment:
Theoretically if all of the components properly handle preserving indices including the label encoder then no... but I figured it would be good to have on the pipelines just in case!